packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -1244,3 +1254,173 @@ export function validateEventListenerTarget(
   }
   return false;
 }
+
+export const supportsTestSelectors = true;
+
+export function findFiberRoot(node: Instance): null | FiberRoot {
+  const stack = [node];
+  let index = 0;
+  while (index < stack.length) {
+    const current = stack[index++];
+    if (isContainerMarkedAsRoot(current)) {
+      return ((getInstanceFromNodeDOMTree(current): any): FiberRoot);
+    }
+    stack.push(...current.children);
+  }
+  return null;
+}
+
+export function getBoundingRect(node: Instance): BoundingRect {
+  const rect = node.getBoundingClientRect();
+  return {
+    x: rect.left,
+    y: rect.top,
+    width: rect.width,
+    height: rect.height,
+  };
+}
+
+export function matchAccessibilityRole(node: Instance, role: string): boolean {
+  if (role === getRole(node)) {
+    return true;
+  }
+
+  return false;
+}
+
+export function getTextContent(fiber: Fiber): string | null {
+  switch (fiber.tag) {
+    case HostComponent:
+      let textContent = '';
+      const childNodes = fiber.stateNode.childNodes;
+      for (let i = 0; i < childNodes.length; i++) {
+        const childNode = childNodes[i];
+        if (childNode.nodeType === Node.TEXT_NODE) {
+          textContent += childNode.textContent;
+        }
+      }
+      return textContent;
+    case HostText:
+      return fiber.stateNode.textContent;
+  }
+
+  return null;
+}
+
+export function isHiddenSubtree(fiber: Fiber): boolean {
+  return fiber.tag === HostComponent && fiber.memoizedProps.hidden === true;
+}
+
+export function setFocusIfFocusable(node: Instance): boolean {
+  if (node.nodeType !== Node.ELEMENT_NODE) {
+    // Text and comment nodes aren't focusable.
+    // Technically this check should not be necessary,
+    // since React treats elements (Instances) and text (TextInstance) differently.
+    return false;
+  }
+
+  if (((node: any): HTMLInputElement).disabled === true) {
+    // Disabled inputs can't be focused.
+    return false;
+  }
+
+  const element = ((node: any): HTMLElement);
+
+  if (element.tabIndex === null || element.tabIndex < 0) {
+    // The HTML spec says that negative tab index values indicate an element should be,
+    // "click focusable but not sequentially focusable".
```

Comment:
Should the function name reflect that? Negative tab indices are also used to programmatically move focus.